src/libsync/vfs/cfapi/cfapiwrapper.cpp

```diff
@@ -694,7 +700,7 @@ OCC::Result<void, QString> OCC::CfApiWrapper::convertToPlaceholder(const FileHan
 
     const auto fileIdentity = QString::fromUtf8(fileId).toStdWString();
     const auto fileIdentitySize = (fileIdentity.length() + 1) * sizeof(wchar_t);
-    const qint64 result = CfConvertToPlaceholder(handle.get(), fileIdentity.data(), sizeToDWORD(fileIdentitySize), CF_CONVERT_FLAG_NONE, nullptr, nullptr);
+    const qint64 result = CfConvertToPlaceholder(handle.get(), fileIdentity.data(), sizeToDWORD(fileIdentitySize), CF_CONVERT_FLAG_MARK_IN_SYNC, nullptr, nullptr);
```

Comment:
That fills a bit heavy handed to me since it'll mark it sync indiscriminately all placeholders which I guess it might not always be the case... or my memory of that API gets fuzzier now. Isn't it something we'd rather deal with from `VfsCfApi::fileStatusChanged`?

Currently its implementation is empty because in my tests I couldn't find anything really wrong with having CF_CONVERT_FLAG_NONE and nothing more. But apparently I was looking at the wrong place.

Reply:
As far as I understand, HydrationJob are implied when some software (via the user) wants to access a file that has not been hydrated.

If the file is being downloaded because it has been pinned, we do not use the HydrationJob but we always go through `CfConvertToPlaceholder`.

I did *some* tests with the change and could not notice issues. I can try harder to find issues.

Reply:
> As far as I understand, HydrationJob are used when some software (via the user) wants to access a file that has not been hydrated.

Right and so this should handle the flag change for us I guess?

> If the file is being downloaded because it has been pinned, we do not use the HydrationJob but we always go through `CfConvertToPlaceholder`.

Indeed. Is it all files or only for files which are unavailable? I admit I don't quite remember... That's also where I'm not sure if CF_CONVERT_FLAG_MARK_IN_SYNC is really appropriate in all cases. Because there are the transient states like "it needs a sync" and "it's syncing". I wonder how that plays along with those.

As you can tell I'm mostly brain dumping here. :-)

> I did _some_ tests with the change and could not notice issues. I can try harder to find issues.

I'd look around transient states mostly (makes them harder to test though). Might not be a problem in practice.

Reply:
@er-vin Documentation for this API feels incomplete, really. From my tests on Matthiue's branch, this status icon's state behaves much better now.

Reply:
> > As far as I understand, HydrationJob are used when some software (via the user) wants to access a file that has not been hydrated.
> 
> Right and so this should handle the flag change for us I guess?

And this part is showing the correct status. I had just needed to avoid having the progress bar reappear when the download is completed.

> 
> > If the file is being downloaded because it has been pinned, we do not use the HydrationJob but we always go through `CfConvertToPlaceholder`.
> 
> Indeed. Is it all files or only for files which are unavailable? I admit I don't quite remember... That's also where I'm not sure if CF_CONVERT_FLAG_MARK_IN_SYNC is really appropriate in all cases. Because there are the transient states like "it needs a sync" and "it's syncing". I wonder how that plays along with those.

If I understand what happen, we have:

1) the user modifies the pinning state of some files that were cloud only (not hydrated)
2) our directory watcher detects that and we see that files have to be downloaded
3) we convert the plain old files into placeholders via `CfConvertToPlaceholder`

I suspect that indeed I have not tested enough what happen if a files is half downloaded. I suppose that we already have quite some code handling that given we are just using the pre-existing code path to do that. This is just after completeion that we have special handling of CfApi.

> 
> As you can tell I'm mostly brain dumping here. :-)
> 
> > I did _some_ tests with the change and could not notice issues. I can try harder to find issues.
> 
> I'd look around transient states mostly (makes them harder to test though). Might not be a problem in practice.

I will. Thanks for the suggestion

Reply:
I did more tests and could only get an error when trying implicit hydration at the same time files where being downloaded by pinning.
At all time, it looks like file status were correct.

Unrelated note: When trying to open the same file twice, I got two download and the second one terminated with an error.

Reply:
@mgallien Would be good to detect "this file is already being hydrated" and then - early-out from the second hydration attempt.

Reply:
Implicit hydration of the same file two times is not triggering any error and is behaving like OneDrive. I believe we are OK there.